diff -ruwN source/meson.build source-new/meson.build
--- source/meson.build	2025-12-29 17:51:03.000000000 +0100
+++ source-new/meson.build	2026-05-22 05:50:54.280346791 +0200
@@ -47,8 +47,6 @@
 libxfce4windowing = dependency('libxfce4windowing-0', version: dependency_versions['libxfce4windowing'])
 xfconf = dependency('libxfconf-0', version: dependency_versions['xfce4'])
 
-gdk_pixbuf_csource = find_program(gdk_pixbuf.get_variable(pkgconfig: 'gdk_pixbuf_csource'), required: true)
-
 # Feature: 'x11'
 x11_deps = []
 x11_deps += dependency('gdk-x11-3.0', version: dependency_versions['gtk'], required: get_option('x11'))
@@ -75,8 +73,6 @@
     feature_cflags += '-DHAVE__ICETRANSNOLISTEN=1'
   endif
 
-  iceauth = find_program('iceauth', required: true)
-
   xsession_prefix = get_option('xsession-prefix')
   if xsession_prefix == ''
     xsession_prefix = '/usr'
diff -ruwN source/xfce4-session/meson.build source-new/xfce4-session/meson.build
--- source/xfce4-session/meson.build	2025-12-29 17:51:03.000000000 +0100
+++ source-new/xfce4-session/meson.build	2026-05-22 05:51:20.133774608 +0200
@@ -42,7 +42,7 @@
 
 if enable_x11
   c_args += [
-    '-DICEAUTH_CMD="@0@"'.format(iceauth.full_path()),
+    '-DICEAUTH_CMD="@0@"'.format('/usr/bin/iceauth'),
   ]
 
   session_sources += [
@@ -89,7 +89,7 @@
   'xfsm-chooser-icon.h',
   input: 'xfsm-chooser-icon.png',
   output: 'xfsm-chooser-icon.h',
-  command: [gdk_pixbuf_csource, '--static', '--raw', '--stream', '--name=xfsm_chooser_icon_data', '@INPUT@'],
+  command: ['gdk-pixbuf-csource', '--static', '--raw', '--stream', '--name=xfsm_chooser_icon_data', '@INPUT@'],
   capture: true,
 )
 
diff -ruwN source/xfce4-session/xfsm-manager.h source-new/xfce4-session/xfsm-manager.h
--- source/xfce4-session/xfsm-manager.h	2025-12-29 17:51:03.000000000 +0100
+++ source-new/xfce4-session/xfsm-manager.h	2026-05-22 05:43:17.663128999 +0200
@@ -26,6 +26,7 @@
 #include <glib-object.h>
 #include <libxfce4util/libxfce4util.h>
 #include <xfconf/xfconf.h>
+#include <strings.h>
 
 #include "xfsm-client.h"
 #include "xfsm-global.h"
