diff -ruw source/client/servers.c source-new/client/servers.c
--- source/client/servers.c	2018-07-21 15:42:11.000000000 -0600
+++ source-new/client/servers.c	2018-12-10 09:31:49.817815930 -0700
@@ -80,6 +80,14 @@
 
 #include "gui_main_g.h"
 
+// Redox patches
+#define IP_ADD_MEMBERSHIP 35
+#define IP_MULTICAST_TTL 33
+struct ip_mreq {
+  struct in_addr imr_multiaddr;   /* IP multicast address of group */
+  struct in_addr imr_interface;   /* local IP address of interface */
+};
+
 struct server_scan {
   enum server_scan_type type;
   ServerScanErrorFunc error_func;
