diff --color -ruwN source/configure source-new/configure
--- source/configure	2026-04-14 15:22:14.000000000 +0700
+++ source-new/configure	2026-05-25 18:24:23.013642280 +0700
@@ -6862,7 +6862,7 @@
   ;;
 
 # This must be glibc/ELF.
-linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
+linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu* | redox*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -13006,7 +13006,7 @@
   ;;
 
 # This must be glibc/ELF.
-linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
+linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu* | redox*)
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
