diff -rupNw source-original/src/io/gfx/video.cpp source/src/io/gfx/video.cpp
--- source-original/src/io/gfx/video.cpp	2018-12-03 01:26:03.479047234 +0100
+++ source/src/io/gfx/video.cpp	2018-12-02 21:50:10.650021649 +0100
@@ -516,7 +516,7 @@ void Video::flip (int mspf, PaletteEffec
 	}
 
 	// Show what has been drawn
-	SDL_Flip(screen);
+	SDL_UpdateRect(screen, 0, 0, screen->w, screen->h);
 
 	return;
 
