diff -ruwN source/configure source-new/configure
--- source/configure	2022-11-06 04:13:26.000000000 +0700
+++ source-new/configure	2025-09-17 21:14:28.163192730 +0700
@@ -6386,7 +6386,7 @@
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
-	(linux*|gnu*|k*bsd*-gnu)
+	(linux*|gnu*|k*bsd*-gnu|redox*)
 		if test "$DFT_LWR_MODEL" = "shared" && test -n "$LD_RPATH_OPT" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
