diff -ruwN source/src/duk_config.h source-new/src/duk_config.h
--- source/src/duk_config.h	2017-09-22 19:06:08.000000000 -0600
+++ source-new/src/duk_config.h	2022-11-12 08:05:16.384212164 -0700
@@ -786,7 +786,7 @@
 /* --- Generic UNIX --- */
 #define DUK_USE_DATE_NOW_GETTIMEOFDAY
 #define DUK_USE_DATE_TZO_GMTIME_R
-#define DUK_USE_DATE_PRS_STRPTIME
+#undef DUK_USE_DATE_PRS_STRPTIME
 #define DUK_USE_DATE_FMT_STRFTIME
 #include <time.h>
 #include <sys/time.h>
