diff -ru make-4.2/config/config.sub make-4.2-new/config/config.sub
--- make-4.2/config/config.sub	2018-11-16 20:15:41.363343886 -0700
+++ make-4.2-new/config/config.sub	2018-11-16 20:15:58.095419089 -0700
@@ -1376,7 +1376,7 @@
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
-	      | -sym* | -kopensolaris* | -plan9* \
+	      | -sym* | -kopensolaris* | -plan9* | -redox* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* | -cloudabi* | -sortix* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff -ru make-4.2/getopt1.c make-4.2-new/getopt1.c
--- make-4.2/getopt1.c	2018-11-16 20:15:41.379343966 -0700
+++ make-4.2-new/getopt1.c	2018-11-16 20:28:25.520961828 -0700
@@ -48,6 +48,8 @@
 #endif
 #endif
 
+#define ELIDE_CODE
+
 #ifndef ELIDE_CODE
 
 
diff -ru make-4.2/getopt.c make-4.2-new/getopt.c
--- make-4.2/getopt.c	2018-11-16 20:15:41.359343867 -0700
+++ make-4.2-new/getopt.c	2018-11-16 20:28:07.653254603 -0700
@@ -56,6 +56,8 @@
 # endif
 #endif
 
+#define ELIDE_CODE
+
 #ifndef ELIDE_CODE
 
 
diff -ru make-4.2/glob/glob.c make-4.2-new/glob/glob.c
--- make-4.2/glob/glob.c	2018-11-16 20:15:41.359343867 -0700
+++ make-4.2-new/glob/glob.c	2018-11-16 20:15:58.095419089 -0700
@@ -625,9 +625,9 @@
 #else
           const char *home_dir = getenv ("HOME");
 #endif
-# ifdef _AMIGA
+# if 1
 	  if (home_dir == NULL || home_dir[0] == '\0')
-	    home_dir = "SYS:";
+	    home_dir = "/home/user";
 # else
 #  ifdef WINDOWS32
 	  if (home_dir == NULL || home_dir[0] == '\0')
@@ -717,7 +717,7 @@
 	      dirname = newp;
 	    }
 	}
-# if !defined _AMIGA && !defined WINDOWS32 && !defined VMS
+# if 0
       else
 	{
 	  char *end_name = strchr (dirname, '/');
diff -ru make-4.2/job.c make-4.2-new/job.c
--- make-4.2/job.c	2018-11-16 20:15:41.351343827 -0700
+++ make-4.2-new/job.c	2018-11-16 20:15:58.099419106 -0700
@@ -1038,7 +1038,7 @@
 void
 block_sigs (void)
 {
-#ifdef POSIX
+#if 0
   (void) sigprocmask (SIG_BLOCK, &fatal_signal_set, (sigset_t *) 0);
 #else
 # ifdef HAVE_SIGSETMASK
@@ -1051,9 +1051,11 @@
 void
 unblock_sigs (void)
 {
+/*
   sigset_t empty;
   sigemptyset (&empty);
   sigprocmask (SIG_SETMASK, &empty, (sigset_t *) 0);
+*/
 }
 #endif
 
diff -ru make-4.2/read.c make-4.2-new/read.c
--- make-4.2/read.c	2018-11-16 20:15:41.359343867 -0700
+++ make-4.2-new/read.c	2018-11-16 20:15:58.099419106 -0700
@@ -2937,7 +2937,7 @@
           free (home_dir);
           home_dir = getenv ("HOME");
         }
-# if !defined(_AMIGA) && !defined(WINDOWS32)
+# if 0
       if (home_dir == 0 || home_dir[0] == '\0')
         {
           char *logname = getlogin ();
@@ -2958,7 +2958,7 @@
           return new;
         }
     }
-# if !defined(_AMIGA) && !defined(WINDOWS32)
+# if 0
   else
     {
       struct passwd *pwent;
