diff -ruwN SDL_mixer-1.2.12/build-scripts/config.sub source/build-scripts/config.sub
--- SDL_mixer-1.2.12/build-scripts/config.sub	2012-01-15 15:01:05.000000000 -0700
+++ source/build-scripts/config.sub	2026-05-19 18:09:33.341873031 -0600
@@ -1276,7 +1276,7 @@
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
-	      | -kopensolaris* \
+	      | -kopensolaris* | -redox* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff -ruwN SDL_mixer-1.2.12/timidity/config.h source/timidity/config.h
--- SDL_mixer-1.2.12/timidity/config.h	2012-01-15 15:01:05.000000000 -0700
+++ source/timidity/config.h	2026-05-19 18:09:33.342251993 -0600
@@ -162,6 +162,7 @@
    then specify the library directory in the configuration file. */
 #define CONFIG_FILE	"timidity.cfg"
 #define CONFIG_FILE_ETC "/etc/timidity.cfg"
+#define CONFIG_FILE_ETC_TIMIDITY_FREEPATS "/etc/timidity/freepats.cfg"
 
 #if defined(__WIN32__) || defined(__OS2__)
 #define DEFAULT_PATH	"C:\\TIMIDITY"
diff -ruwN SDL_mixer-1.2.12/timidity/timidity.c source/timidity/timidity.c
--- SDL_mixer-1.2.12/timidity/timidity.c	2012-01-15 15:01:05.000000000 -0700
+++ source/timidity/timidity.c	2026-05-19 18:09:33.342399700 -0600
@@ -286,10 +286,12 @@
   if (!env || read_config_file(env)<0) {
     if (read_config_file(CONFIG_FILE)<0) {
       if (read_config_file(CONFIG_FILE_ETC)<0) {
+        if (read_config_file(CONFIG_FILE_ETC_TIMIDITY_FREEPATS)<0) {
         return(-1);
       }
     }
   }
+  }
 
   if (channels < 1 || channels == 3 || channels == 5 || channels > 6) return(-1);
 
