diff -rupN source/src/timer/unix/SDL_systimer.c source-redox/src/timer/unix/SDL_systimer.c
--- source/src/timer/unix/SDL_systimer.c	2012-01-19 07:30:06.000000000 +0100
+++ source-redox/src/timer/unix/SDL_systimer.c	2018-04-22 17:42:17.455342646 +0200
@@ -150,14 +150,14 @@ static void HandleAlarm(int sig)
 
 int SDL_SYS_TimerInit(void)
 {
-	struct sigaction action;
+	//struct sigaction action;
 
-	/* Set the alarm handler (Linux specific) */
-	SDL_memset(&action, 0, sizeof(action));
-	action.sa_handler = HandleAlarm;
-	action.sa_flags = SA_RESTART;
-	sigemptyset(&action.sa_mask);
-	sigaction(SIGALRM, &action, NULL);
+	///* Set the alarm handler (Linux specific) */
+	//SDL_memset(&action, 0, sizeof(action));
+	//action.sa_handler = HandleAlarm;
+	//action.sa_flags = SA_RESTART;
+	//sigemptyset(&action.sa_mask);
+	//sigaction(SIGALRM, &action, NULL);
 	return(0);
 }
 
@@ -168,22 +168,22 @@ void SDL_SYS_TimerQuit(void)
 
 int SDL_SYS_StartTimer(void)
 {
-	struct itimerval timer;
+	//struct itimerval timer;
 
-	timer.it_value.tv_sec = (SDL_alarm_interval/1000);
-	timer.it_value.tv_usec = (SDL_alarm_interval%1000)*1000;
-	timer.it_interval.tv_sec = (SDL_alarm_interval/1000);
-	timer.it_interval.tv_usec = (SDL_alarm_interval%1000)*1000;
-	setitimer(ITIMER_REAL, &timer, NULL);
+	//timer.it_value.tv_sec = (SDL_alarm_interval/1000);
+	//timer.it_value.tv_usec = (SDL_alarm_interval%1000)*1000;
+	//timer.it_interval.tv_sec = (SDL_alarm_interval/1000);
+	//timer.it_interval.tv_usec = (SDL_alarm_interval%1000)*1000;
+	//setitimer(ITIMER_REAL, &timer, NULL);
 	return(0);
 }
 
 void SDL_SYS_StopTimer(void)
 {
-	struct itimerval timer;
+	//struct itimerval timer;
 
-	SDL_memset(&timer, 0, (sizeof timer));
-	setitimer(ITIMER_REAL, &timer, NULL);
+	//SDL_memset(&timer, 0, (sizeof timer));
+	//setitimer(ITIMER_REAL, &timer, NULL);
 }
 
 #else /* USE_ITIMER */
