diff -rupNw source-original/configure.in source/configure.in
--- source-original/configure.in	2012-01-19 07:30:05.000000000 +0100
+++ source/configure.in	2018-05-03 00:23:39.060438383 +0200
@@ -2445,6 +2445,11 @@ case "$host" in
                 SOURCES="$SOURCES $srcdir/src/cdrom/osf/*.c"
                 have_cdrom=yes
             ;;
+            redox)
+                AC_DEFINE(SDL_CDROM_DUMMY)
+                SOURCES="$SOURCES $srcdir/src/cdrom/dummy/*.c"
+                have_cdrom=yes
+            ;;
           esac
         fi
         # Set up files for the thread library
