--- source/make-ca.sh	2018-08-25 17:38:12.512463896 +0200
+++ build/make-ca.sh	2018-09-23 14:45:12.492919202 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Begin /usr/sbin/make-ca.sh
 #
 # Script to create OpenSSL certs directory, GnuTLS certificate bundle, NSS
@@ -28,9 +28,9 @@
 CERTDATA="certdata.txt"
 PKIDIR="/etc/pki"
 SSLDIR="/etc/ssl"
-CERTUTIL="/usr/bin/certutil"
-KEYTOOL="/opt/jdk/bin/keytool"
-OPENSSL="/usr/bin/openssl"
+CERTUTIL="certutil"
+KEYTOOL="keytool"
+OPENSSL="openssl"
 ANCHORDIR="${PKIDIR}/anchors"
 CABUNDLE="${SSLDIR}/ca-bundle.crt"
 CERTDIR="${SSLDIR}/certs"
@@ -723,7 +723,7 @@
   unset cert
 fi
 
-/usr/bin/c_rehash "${DESTDIR}${CERTDIR}" 2>&1>/dev/null
+c_rehash "${DESTDIR}${CERTDIR}" 2>&1>/dev/null
 popd > /dev/null
 
 # Clean up the mess
