From: bjorn3 <17426603+bjorn3@users.noreply.github.com>
Date: Sat, 6 Dec 2025 15:59:20 +0100
Subject: [PATCH] Fix compilation on Redox OS

---
 include/urcu/syscall-compat.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/urcu/syscall-compat.h b/include/urcu/syscall-compat.h
index 23b266e..2c3b03f 100644
--- a/include/urcu/syscall-compat.h
+++ b/include/urcu/syscall-compat.h
@@ -33,7 +33,8 @@
 #include <syscall.h>
 
 #elif defined(__CYGWIN__) || defined(__APPLE__) || \
-	defined(__FreeBSD__) || defined(__DragonFly__)
+	defined(__FreeBSD__) || defined(__DragonFly__) || \
+	defined(__redox__)
 /* Don't include anything on these platforms. */
 
 #else
-- 
2.47.3

