diff -ruwN source/src/memfile.c source-new/src/memfile.c
--- source/src/memfile.c	2016-11-10 10:24:37.000000000 -0700
+++ source-new/src/memfile.c	2020-01-25 18:19:55.524324451 -0700
@@ -615,6 +615,8 @@
 	    /* No sync() on Stratus VOS */
 # if defined(__OPENNT) || defined(__TANDEM) || defined(__VOS__)
 	    fflush(NULL);
+# elif defined(__redox__)
+        fsync(mfp->mf_fd);
 # else
 	    sync();
 # endif
