diff -ru source/src/common/tuklib_open_stdxxx.c source-new/src/common/tuklib_open_stdxxx.c
--- source/src/common/tuklib_open_stdxxx.c	2016-12-30 03:08:20.000000000 -0800
+++ source-new/src/common/tuklib_open_stdxxx.c	2017-08-08 12:38:10.703829409 -0700
@@ -35,8 +35,8 @@
 			// writing to stdin would fail. However, /dev/full
 			// is Linux specific, and if the program tries to
 			// write to stdin, there's already a problem anyway.
-			const int fd = open("/dev/null", O_NOCTTY
-					| (i == 0 ? O_WRONLY : O_RDONLY));
+			const int fd = open("/dev/null",
+					(i == 0 ? O_WRONLY : O_RDONLY));
 
 			if (fd != i) {
 				if (fd != -1)
