diff -ru source/src/xz/message.c source-new/src/xz/message.c
--- source/src/xz/message.c	2016-12-30 03:08:20.000000000 -0800
+++ source-new/src/xz/message.c	2017-08-08 12:44:30.584447139 -0700
@@ -135,7 +135,7 @@
 	}
 */
 
-#ifdef SIGALRM
+#if 0
 	// Establish the signal handlers which set a flag to tell us that
 	// progress info should be updated.
 	struct sigaction sa;
diff -ru source/src/xz/signals.c source-new/src/xz/signals.c
--- source/src/xz/signals.c	2016-12-30 03:08:20.000000000 -0800
+++ source-new/src/xz/signals.c	2017-08-08 12:41:59.624111395 -0700
@@ -53,6 +53,7 @@
 extern void
 signals_init(void)
 {
+#if 0
 	// List of signals for which we establish the signal handler.
 	static const int sigs[] = {
 		SIGINT,
@@ -112,6 +113,7 @@
 	signals_are_initialized = true;
 
 	return;
+#endif
 }
 
 
@@ -152,6 +154,7 @@
 extern void
 signals_exit(void)
 {
+#if 0
 	const int sig = exit_signal;
 
 	if (sig != 0) {
@@ -171,6 +174,7 @@
 	}
 
 	return;
+#endif
 }
 
 #else
