diff --git a/src/uucore/src/lib/mods/locale.rs b/src/uucore/src/lib/mods/locale.rs
index 60ca9d42..31f3ba0b 100644
--- a/src/uucore/src/lib/mods/locale.rs
+++ b/src/uucore/src/lib/mods/locale.rs
@@ -317,10 +317,10 @@ fn init_localization(
         }
     };
 
-    LOCALIZER.with(|lock| {
+    let _ = LOCALIZER.with(|lock| {
         lock.set(loc)
             .map_err(|_| LocalizationError::Bundle("Localizer already initialized".into()))
-    })?;
+    });
     Ok(())
 }
 
@@ -637,10 +637,10 @@ pub fn setup_localization(p: &str) -> Result<(), LocalizationError> {
             Localizer::new(english_bundle)
         };
 
-        LOCALIZER.with(|lock| {
+        let _ = LOCALIZER.with(|lock| {
             lock.set(localizer)
                 .map_err(|_| LocalizationError::Bundle("Localizer already initialized".into()))
-        })?;
+        });
     }
     LOCALIZER_IS_SET.with(|f| f.set(true));
     Ok(())
@@ -919,10 +919,10 @@ mod tests {
             }
         };
 
-        LOCALIZER.with(|lock| {
+        let _ = LOCALIZER.with(|lock| {
             lock.set(loc)
                 .map_err(|_| LocalizationError::Bundle("Localizer already initialized".into()))
-        })?;
+        });
         Ok(())
     }
 
