--- source/game.cpp	2018-06-17 18:09:41.545548000 +0200
+++ build/game.cpp	2018-06-17 20:06:47.516838000 +0200
@@ -2900,8 +2900,8 @@
 
 #if SDL_MAJOR_VERSION == 1
 		const SDL_VideoInfo *videoInfo = SDL_GetVideoInfo();
-		*user_width = videoInfo->current_w;
-		*user_height = videoInfo->current_h;
+		*user_width = 640;//videoInfo->current_w;
+		*user_height = 480;//videoInfo->current_h;
 		LOG("desktop is %d x %d\n", *user_width, *user_height);
 #else
 		SDL_DisplayMode displayMode;
@@ -2942,11 +2942,11 @@
 		//user_width = 1184;
 		//user_height = 720;
 
-		if( user_width >= 4*default_width_c ) {
+		/*if( user_width >= 4*default_width_c ) {
 			scale_width = 4.0f;
 			LOG("scale width 4x\n");
 		}
-		else if( user_width >= 3*default_width_c ) {
+		else*/ if( user_width >= 3*default_width_c ) {
 			scale_width = 3.0f;
 			LOG("scale width 3x\n");
 		}
@@ -2963,11 +2963,11 @@
 			return false;
 		}
 
-		if( user_height >= 4*default_height_c ) {
+		/*if( user_height >= 4*default_height_c ) {
 			scale_height = 4.0f;
 			LOG("scale height 4x\n");
 		}
-		else if( user_height >= 3*default_height_c ) {
+		else*/ if( user_height >= 3*default_height_c ) {
 			scale_height = 3.0f;
 			LOG("scale height 3x\n");
 		}
@@ -3022,11 +3022,11 @@
 		screen = new Gigalomania::Screen();
 
 #if SDL_MAJOR_VERSION == 1
-		if( screen->open(4*default_width_c, 4*default_height_c, fullscreen) ) {
+		/*if( screen->open(4*default_width_c, 4*default_height_c, fullscreen) ) {
 			scale_width = scale_height = 4.0f;
 			LOG("scale 4x\n");
 		}
-		else if( screen->open(3*default_width_c, 3*default_height_c, fullscreen) ) {
+		else*/ if( screen->open(3*default_width_c, 3*default_height_c, fullscreen) ) {
 			scale_width = scale_height = 3.0f;
 			LOG("scale 3x\n");
 		}
@@ -3636,7 +3636,7 @@
 }
 
 //bool quit = false;
-bool debugwindow = false;
+bool debugwindow = true;
 
 void Game::requestQuit(bool force_quit) {
     if( !state_changed ) {
@@ -5243,7 +5243,7 @@
 	debugwindow = true;
 #endif
 
-	bool fullscreen = true;
+	bool fullscreen = false;
 #if defined(__amigaos4__) || defined(AROS) || defined(__MORPHOS__)
 	fullscreen = false; // run in windowed mode due to reported performance problems in fullscreen mode on AmigaOS 4; also randomly hangs on AROS in fullscreen mode; also included MorphOS just to be safe
 #endif
