--- source/gamestate.cpp	2018-05-28 11:51:01.789562777 +0200
+++ build/gamestate.cpp	2018-05-28 14:24:06.690031507 +0200
@@ -1428,8 +1428,9 @@
 };
 
 void GameState::fadeScreen(bool out, int delay, void (*func_finish)()) {
-    if( fade != NULL )
+    if( fade != NULL ) {
         delete fade;
+	}
 	if( game_g->isTesting() ) {
 		if( func_finish != NULL ) {
 			func_finish();
@@ -1442,8 +1443,9 @@
 
 void GameState::whiteFlash() {
 	//ASSERT( whitefade == NULL );
-    if( whitefade != NULL )
+    if( whitefade != NULL ) {
         delete whitefade;
+	}
 	if( !game_g->isTesting() ) {
 	    whitefade = new FadeEffect(true, false, 0, NULL);
 	}
