--- a/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c	2018-06-03 07:26:17.413729023 +0200
+++ b/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c	2018-06-03 07:41:40.119390067 +0200
@@ -162,10 +162,7 @@
 /* LCOV_EXCL_START */
     struct stat        st;
     static const char *devices[] = {
-# ifndef USE_BLOCKING_RANDOM
-        "/dev/urandom",
-# endif
-        "/dev/random", NULL
+        "rand:", NULL
     };
     const char       **device = devices;
     int                fd;
