diff -burpN source-original/configure.ac source/configure.ac
--- source-original/configure.ac	2008-11-09 20:12:37.000000000 +0100
+++ source/configure.ac	2018-04-22 23:41:16.945896818 +0200
@@ -85,8 +85,6 @@ if test "$cross_compiling" != "yes"; the
 fi
 
 dnl --- Header files, typedefs, structures
-AC_TYPE_UID_T
-AC_TYPE_SIZE_T
 AC_DECL_SYS_SIGLIST
 AC_HEADER_SYS_WAIT
 AC_CHECK_HEADERS(unistd.h asm/byteorder.h sched.h)
