diff --color -ruwN source/meson.build source-new/meson.build
--- source/meson.build	2023-03-22 19:13:43.000000000 +0700
+++ source-new/meson.build	2026-06-14 14:55:40.778818599 +0700
@@ -99,7 +99,7 @@
 endif
 
 dep_glx = dependency('glx', required: false, disabler : true)
-if not dep_glx.found() and host_machine.system() == 'darwin'
+if not dep_glx.found()
   # xquartz doesn't have a glx.pc, but it does have the header. And all the
   # symbols reside in libGL, so let's just use that.
   if cc.check_header('GL/glx.h', dependencies: dep_x11)
