diff -ruwN source/meson.build source-new/meson.build
--- source/meson.build	2024-03-24 04:23:43.000000000 +0700
+++ source-new/meson.build	2025-09-30 17:40:34.433925985 +0700
@@ -637,8 +637,6 @@
     libxkbcommon_x11_test_internal = static_library(
         'xkbcommon-x11-internal',
         libxkbcommon_x11_sources,
-        'test/xvfb-wrapper.c',
-        'test/xvfb-wrapper.h',
         include_directories: include_directories('src', 'include'),
         link_with: libxkbcommon_test_internal,
         dependencies: [
@@ -768,20 +766,6 @@
         dependencies: test_dep),
     env: test_env,
 )
-if get_option('enable-x11')
-    test(
-        'x11',
-        executable('test-x11', 'test/x11.c', dependencies: x11_test_dep),
-        env: test_env,
-        is_parallel : false,
-    )
-    test(
-        'x11comp',
-        executable('test-x11comp', 'test/x11comp.c', dependencies: x11_test_dep),
-        env: test_env,
-        is_parallel : false,
-    )
-endif
 if get_option('enable-xkbregistry')
     test(
         'registry',
