diff -ruw source/config.sub source-new/config.sub
--- source/config.sub	2012-01-14 21:44:08.000000000 -0700
+++ source-new/config.sub	2018-12-10 12:17:28.136784814 -0700
@@ -1276,7 +1276,7 @@
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
-	      | -kopensolaris* \
+	      | -kopensolaris* | -redox* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
Only in source-new: config.sub.orig
diff -ruw source/configure.in source-new/configure.in
--- source/configure.in	2012-01-14 21:44:08.000000000 -0700
+++ source-new/configure.in	2018-12-10 12:17:12.724883206 -0700
@@ -91,38 +91,13 @@
 AM_CONDITIONAL(USE_VERSION_RC, test x$use_version_rc = xtrue)
 
 dnl Check for the FreeType 2 library
-dnl
-dnl Get the cflags and libraries from the freetype-config script
-dnl
-AC_ARG_WITH(freetype-prefix,[  --with-freetype-prefix=PFX   Prefix where FREETYPE is 
-installed (optional)],
-            freetype_prefix="$withval", freetype_prefix="")
-AC_ARG_WITH(freetype-exec-prefix,[  --with-freetype-exec-prefix=PFX Exec prefix 
-where FREETYPE is installed (optional)],
-            freetype_exec_prefix="$withval", freetype_exec_prefix="")
-
-if test x$freetype_exec_prefix != x ; then
-     freetype_args="$freetype_args --exec-prefix=$freetype_exec_prefix"
-     if test x${FREETYPE_CONFIG+set} != xset ; then
-        FREETYPE_CONFIG=$freetype_exec_prefix/bin/freetype-config
-     fi
-fi
-if test x$freetype_prefix != x ; then
-     freetype_args="$freetype_args --prefix=$freetype_prefix"
-     if test x${FREETYPE_CONFIG+set} != xset ; then
-        FREETYPE_CONFIG=$freetype_prefix/bin/freetype-config
-     fi
-fi
-AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
-no_freetype=""
-if test "$FREETYPE_CONFIG" = "no" ; then
-    AC_MSG_ERROR([
-*** Unable to find FreeType2 library (http://www.freetype.org/)
+#PKG_CHECK_MODULES([FT2], [freetype2 >= 7.0.1], [], [dnl
+#    AC_CHECK_FT2(,,[AC_MSG_ERROR([dnl
+#*** Unable to find FreeType2 library (http://www.freetype.org/)])]
+#    )
+#])
-])
-else
-    CFLAGS="$CFLAGS `$FREETYPE_CONFIG $freetypeconf_args --cflags`"
-    LIBS="$LIBS `$FREETYPE_CONFIG $freetypeconf_args --libs`"
-fi
+CFLAGS="$CFLAGS $FT2_CFLAGS"
+LIBS="$LIBS $FT2_LIBS -lfreetype -lpng -lz"
 
 dnl Check for SDL
 SDL_VERSION=1.2.4
