diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 45e2fc7a..2d7f7db3 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -175,28 +175,32 @@ sub vms_info {
     },
 
 
-### Redox
-    "redox-aarch64" => {
-        inherit_from     => [ "BASE_common", asm("aarch64_asm") ],
+### Redox configurations
+    "redox-common" => {
+        inherit_from     => [ "BASE_common" ],
         cc               => "gcc",
-        cflags           => "-DL_ENDIAN -DOSSL_SSIZE_MAX=LONG_MAX -DNO_SYSLOG -O3 -static -Wall",
+        thread_scheme    => "(unknown)",
+        dso_scheme       => "dlfcn",
+        shared_target    => "linux-shared",
+        shared_cflag     => "-fPIC",
+        shared_ldflag    => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
+        shared_extension => ".so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+    },
+    "redox-aarch64" => {
+        inherit_from     => [ "redox-common", asm("aarch64_asm") ],
+        cflags           => "-DL_ENDIAN -DOSSL_SSIZE_MAX=LONG_MAX -DNO_SYSLOG -O3 -Wall",
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         perlasm_scheme   => "elf",
-        thread_scheme    => "(unknown)",
     },
     "redox-i686" => {
-        inherit_from     => [ "BASE_unix", asm("x86_elf_asm") ],
-        cc               => "gcc",
-        cflags           => "-DL_ENDIAN -DNO_SYSLOG -O3 -static -Wall",
+        inherit_from     => [ "redox-common", asm("x86_elf_asm") ],
+        cflags           => "-DL_ENDIAN -DNO_SYSLOG -O3 -Wall",
         bn_ops           => "BN_LLONG",
-        thread_scheme    => "(unknown)",
     },
     "redox-x86_64" => {
-        inherit_from     => [ "BASE_common", asm("x86_64_asm") ],
-        cc               => "gcc",
-        cflags           => "-DL_ENDIAN -DOSSL_SSIZE_MAX=LONG_MAX -DNO_SYSLOG -O3 -static -Wall",
+        inherit_from     => [ "redox-common", asm("x86_64_asm") ],
+        cflags           => "-DL_ENDIAN -DOSSL_SSIZE_MAX=LONG_MAX -DNO_SYSLOG -O3 -Wall",
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
-        thread_scheme    => "(unknown)",
     },
 
 #### Solaris configurations
