diff -ruwN source/lib/fseterr.c source-new/lib/fseterr.c
--- source/lib/fseterr.c	2026-03-08 20:32:36.000000000 +0700
+++ source-new/lib/fseterr.c	2026-05-20 10:00:55.592770876 +0700
@@ -53,7 +53,7 @@
 #elif defined EPLAN9                /* Plan9 */
   if (fp->state != 0 /* CLOSED */)
     fp->state = 5 /* ERR */;
-#elif 0                             /* unknown  */
+#elif defined __redox__                /* unknown  */
   /* Portable fallback, based on an idea by Rich Felker.
      Wow! 6 system calls for something that is just a bit operation!
      Not activated on any system, because there is no way to repair FP when
diff -ruwN source/Makefile.in source-new/Makefile.in
--- source/Makefile.in	2026-05-20 10:25:52.529450320 +0700
+++ source-new/Makefile.in	2026-05-20 10:23:19.309839883 +0700
@@ -5875,7 +5875,6 @@
 #
 # Use '-K utf8', not '-k', in case 'configure' didn't find uchardet.
 MOMPDFMOM = \
-  GROFF_COMMAND=test-groff \
   GROFF_COMMAND_PREFIX= \
   GROFF_BIN_PATH="$(GROFF_BIN_PATH)" \
   $(PDFMOMBIN) $(FFLAG) $(MFLAG) -M$(mom_srcdir) -K utf8 -p -e -t \
@@ -6018,7 +6017,6 @@
 @HAVE_URW_FONTS_TRUE@pdfmomoption = -P-W
 @HAVE_URW_FONTS_TRUE@groffmanpagesoption = -P-e
 DOC_PDFMOM = \
-  GROFF_COMMAND=test-groff \
   GROFF_COMMAND_PREFIX= \
   GROFF_BIN_PATH="$(GROFF_BIN_PATH)" \
   $(PDFMOMBIN) $(MFLAG) $(FFLAG) -ww -b $(pdfmomoption) --roff
@@ -6086,7 +6084,7 @@
 PROCESSEDFILES_DEPS_PS = groff troff grops tmac/ps.tmac \
   font/devps/stamp
 
-PROCESSEDFILES_DEPS_PDF = test-groff groff troff gropdf tmac/pdf.tmac \
+PROCESSEDFILES_DEPS_PDF = groff troff gropdf tmac/pdf.tmac \
   font/devpdf/stamp
 
 PROCESSEDFILES_DEPS_TXT = groff troff grotty tmac/tty.tmac \
@@ -19226,7 +19224,7 @@
 	&& LC_ALL=C $(MOMPDFMOM) $< >$@
 
 $(MOMPROCESSEDEXAMPLEFILES): $(MOMNORMALFILES) \
-  test-groff groff troff gropdf pdfmom penguin.ps penguin.pdf \
+  groff troff gropdf pdfmom penguin.ps penguin.pdf \
   font/devpdf/stamp tmac/pdf.tmac
 
 penguin.ps:
