diff -ruwN source/lib/fseterr.c source-new/lib/fseterr.c
--- source/lib/fseterr.c	2026-03-08 20:32:36.000000000 +0700
+++ source-new/lib/fseterr.c	2026-06-16 16:02:23.420661088 +0700
@@ -53,7 +53,7 @@
 #elif defined EPLAN9                /* Plan9 */
   if (fp->state != 0 /* CLOSED */)
     fp->state = 5 /* ERR */;
-#elif 0                             /* unknown  */
+#elif defined __redox__             /* unknown  */
   /* Portable fallback, based on an idea by Rich Felker.
      Wow! 6 system calls for something that is just a bit operation!
      Not activated on any system, because there is no way to repair FP when
diff -ruwN source/test-groff.in source-new/test-groff.in
--- source/test-groff.in	2026-03-14 14:18:16.000000000 +0700
+++ source-new/test-groff.in	2026-06-16 16:01:09.900793444 +0700
@@ -52,4 +52,4 @@
 export GROFF_COMMAND_PREFIX
 export GROFF_BIN_PATH GROFF_FONT_PATH GROFF_TMAC_PATH
 
-exec $builddir/groff "$@"
+#exec $builddir/groff "$@"
